### PR TITLE
[clang][Interp] Only emit function_param_value_unknown in C++11

### PR DESCRIPTION
**Pull request by tbaederr**
This is also what the current interpreter does.

No test for now since the only reproducer I have is from https://reviews.llvm.org/D156565 and doesn't work without that patch.

### CODE REVIEW

**Comment by tbaederr** [comment]
Ping

**Review by AaronBallman — COMMENTED** [review]
Test coverage?

**Comment by AaronBallman** [comment]
> https://reviews.llvm.org/D156565 hasn't been merged as far as I can tell and it's the only test case I have for this. :(

Ah shoot, let me ping that review and see if I can get it accepted and landed so we can have test coverage.

File: clang/test/SemaCXX/offsetof.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // RUN: %clang_cc1 -triple x86_64-apple-darwin10.0.0 -fsyntax-only -verify %s -Winvalid-offsetof -std=c++98
-// RUN: %clang_cc1 -triple x86_64-apple-darwin10.0.0 -fsyntax-only -verify=expected,new-interp %s -Winvalid-offsetof -std=c++98 -fexperimental-new-constant-interpreter
+// RUN: %clang_cc1 -triple x86_64-apple-darwin10.0.0 -fsyntax-only -verify=expected %s -Winvalid-offsetof -std=c++98 -fexperimental-new-constant-interpreter
```
Comment (AaronBallman) [inline clang/test/SemaCXX/offsetof.cpp:2]:
```suggestion
// RUN: %clang_cc1 -triple x86_64-apple-darwin10.0.0 -fsyntax-only -verify %s -Winvalid-offsetof -std=c++98 -fexperimental-new-constant-interpreter
```

**Review by AaronBallman — APPROVED** [review]
LGTM

**Pull request merged**